add inlineJadeRuntime and amdDependencies to options

capricube opened the pull request:
added explicit inlineJadeRuntime parameter to prevent inlining of jade runtime (default true) and an amdDependencies parameter that is used to extend the signature of the define() method

also upgraded jade dependency

lukekarrys commented on README.md:
```diff
@@ -89,6 +89,14 @@ templatizer(templatesDir, 'templates.js', {
+
+By default the jade runtime will be included into the generated template javascript file. In order minimize the file size you can set this parameter to false. Instead a `jade` module is expected as amdDependency parameter. Otherwise an error will be thrown.
+
+#### `dontRemoveMixins` (array, default [])
```
I think this should be `amdDependencies` right?

lukekarrys commented on templatizer.js:
```diff
@@ -35,7 +37,16 @@ module.exports = function (templateDirectories, outputFile, options) {
+    var amdModuleDependencies = '';
+    var separator = '';
+    _.each(options.amdDependencies, function(element) {
+        amdModuleDependencies = separator + '\'' + element + '\'';
```
This looks like it's reassigning `amdModuleDependencies` every time through the loop so that it'll only have the last element. It should probably be `+=` right?

lukekarrys commented:
Hi @capricube, thanks for the PR! I don't have much experience with AMD but I'm happy to include this in the project so others can get the benefit from it. I found a few minor issues with the PR that I've added comments for.

Also the module currently has tests for running the global and browserify versions. Would you be able to add a test for running the AMD version, and for testing the new amd options?

capricube commented:
hi @lukekarrys, thanks for your comments. i'll fix the issues you mentioned and create tests for the new options.

capricube commented:
hi @lukekarrys, i incorporated the changes you suggested and fixed a bug. i also fixed a exisiting browser test. for some reason today it seems impossible for me to fix the merge conflict

capricube commented:
finally fixed merge conflict :)

lukekarrys commented:
@capricube Thanks for your hard work on this! I'll get this published very soon.

lukekarrys commented:
Published as `v1.3.0`

capricube commented:
thank you for merging!
